pkg/polaris/ctl_polaris.go

```diff
@@ -67,317 +66,264 @@ type CRSpecBuilderFromCobraFlags struct {
 	if master {
-		cmd.Flags().StringVar(&ctl.EventstoreSize, "eventstore-size", GetPolarisDefault().PolarisDBSpec.EventstoreDetails.Storage.StorageSize, "Persistent volume claim size for eventstore")
-		cmd.Flags().StringVar(&ctl.MongoDBSize, "mongodb-size", GetPolarisDefault().PolarisDBSpec.MongoDBDetails.Storage.StorageSize, "Persistent volume claim size for mongodb")
-		cmd.Flags().StringVar(&ctl.DownloadServerSize, "downloadserver-size", GetPolarisDefault().PolarisSpec.DownloadServerDetails.Storage.StorageSize, "Persistent volume claim size for downloadserver")
-		cmd.Flags().StringVar(&ctl.UploadServerSize, "uploadserver-size", GetPolarisDefault().PolarisDBSpec.UploadServerDetails.Storage.StorageSize, "Persistent volume claim size for uploadserver")
-		cmd.Flags().StringVar(&ctl.PostgresSize, "postgres-size", GetPolarisDefault().PolarisDBSpec.PostgresDetails.Storage.StorageSize, "Persistent volume claim size to use for postgres. Only applicable if --enable-postgres-container is set to true")
-		cmd.Flags().StringVar(&ctl.StorageClass, "storage-class", ctl.StorageClass, "Set the storage class to use for all persistent volume claims\n")
+		cmd.Flags().StringVar(&ctl.flagTree.EventstoreSize, "eventstore-size", EVENTSTORE_PV_SIZE, "Persistent volume claim size for eventstore")
+		cmd.Flags().StringVar(&ctl.flagTree.MongoDBSize, "mongodb-size", MONGODB_PV_SIZE, "Persistent volume claim size for mongodb")
+		cmd.Flags().StringVar(&ctl.flagTree.DownloadServerSize, "downloadserver-size", DOWNLOAD_SERVER_PV_SIZE, "Persistent volume claim size for download server")
+		cmd.Flags().StringVar(&ctl.flagTree.UploadServerSize, "uploadserver-size", UPLOAD_SERVER_PV_SIZE, "Persistent volume claim size for upload server")
+		cmd.Flags().StringVar(&ctl.flagTree.PostgresSize, "postgres-size", POSTGRES_PV_SIZE, "Persistent volume claim size to use for postgres. Only applicable if --enable-postgres-container is set to true")
+		cmd.Flags().StringVar(&ctl.flagTree.StorageClass, "storage-class", ctl.flagTree.StorageClass, "Set the storage class to use for all persistent volume claims\n")
 	}
 
 	// reporting related flags
-	cmd.Flags().BoolVar(&ctl.EnableReporting, "enable-reporting", GetPolarisDefault().EnableReporting, "Enable Reporting Platform")
-	cmd.Flags().StringVar(&ctl.ReportStorageSize, "reportstorage-size", GetPolarisDefault().ReportingSpec.ReportStorageDetails.Storage.StorageSize, "Persistent volume claim size for reportstorage. Only applicable if --enable-reporting is set to true")
-
-	// flags that are add-ons (helpful, but not required to set up an environment
-	cmd.Flags().StringVar(&ctl.Registry, "registry", ctl.Registry, "Docker registry e.g. docker.io/myuser")
-	cmd.Flags().StringVar(&ctl.ImagePullSecrets, "pull-secret", ctl.ImagePullSecrets, "Pull secret when using a private registry\n")
```

Comment:
fyi Black Duck and Alert are going to support imagePullSecrets

Reply:
In Polaris and Polaris reporting, there is no global registry and also since we need to pass the image pull secret to all sub charts, I have hard coded the name in the chart as 'gcr-key'. Synopsysctl will create the secret with that name.

Reply:
Gotcha cool!